validator/accounts/status.go

```diff
@@ -0,0 +1,240 @@
+func RunStatusCommand(
+	pubkeys [][]byte,
+	withCert string,
+	endpoint string,
+	maxCallRecvMsgSize int,
+	grpcRetries uint,
+	grpcHeaders []string) error {
+	dialOpts, err := constructDialOptions(maxCallRecvMsgSize, withCert, grpcHeaders, grpcRetries)
+	if err != nil {
+		return err
+	}
+	ctx, cancel := context.WithTimeout(
+		context.Background(), 10 * time.Second /* Cancel if cannot connect to beacon node in 10 seconds. */)
+	defer cancel()
+	conn, err := grpc.DialContext(ctx, endpoint, dialOpts...)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to dial beacon node endpoint at %s", endpoint)
+	}
+	statuses, err := FetchAccountStatuses(
+		context.Background(), ethpb.NewBeaconNodeValidatorClient(conn), pubkeys)
+	if e := conn.Close(); e != nil {
+		log.WithError(e).Error("Could not close connection to beacon node")
+	}
+	if err != nil {
+		return errors.Wrap(err, "Could not fetch account statuses from the beacon node")
+	}
+	printStatuses(statuses)
+	return nil
+}
+
+func constructDialOptions(
+	maxCallRecvMsgSize int,
+	withCert string,
+	grpcHeaders []string,
+	grpcRetries uint) ([]grpc.DialOption, error) {
+	var transportSecurity grpc.DialOption
+	if withCert != "" {
+		creds, err := credentials.NewClientTLSFromFile(withCert, "")
+		if err != nil {
+			return nil, errors.Wrapf(err, "Could not get valid credentials: %v", err)
+		}
+		transportSecurity = grpc.WithTransportCredentials(creds)
+	} else {
+		transportSecurity = grpc.WithInsecure()
+		log.Warn(
+			"You are using an insecure gRPC connection! Please provide a certificate and key to use a secure connection.")
+	}
+
+	if maxCallRecvMsgSize == 0 {
+		maxCallRecvMsgSize = 10 * 5 << 20 // Default 50Mb
+	}
+
+	md := make(metadata.MD)
+	for _, hdr := range grpcHeaders {
+		if hdr != "" {
+			ss := strings.Split(hdr, "=")
+			if len(ss) != 2 {
+				log.Warnf("Incorrect gRPC header flag format. Skipping %v", hdr)
+				continue
+			}
+			md.Set(ss[0], ss[1])
+		}
+	}
+
+	return []grpc.DialOption{
+		transportSecurity,
+		grpc.WithDefaultCallOptions(
+			grpc.MaxCallRecvMsgSize(maxCallRecvMsgSize),
+			grpc_retry.WithMax(grpcRetries),
+			grpc.Header(&md),
+		),
+		grpc.WithStatsHandler(&ocgrpc.ClientHandler{}),
+		grpc.WithBlock(),
+	}, nil
+}
+
+// FetchAccountStatuses fetches validator statuses from the BeaconNodeValidatorClient
+// for each validator public key.
+func FetchAccountStatuses(
+	ctx context.Context,
+	beaconNodeRPCProvider ethpb.BeaconNodeValidatorClient,
+	pubkeys [][]byte) ([]ValidatorStatusMetadata, error) {
+	ctx, span := trace.StartSpan(ctx, "accounts.FetchAccountStatuses")
+	defer span.End()
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second /* Cancel if running over thirty seconds. */)
+	defer cancel()
+
+	errorChannel := make(chan error, MaxRequestLimit)
+	statusChannel := make(chan []ValidatorStatusMetadata, MaxRequestLimit)
+	// Launch routines to fetch statuses.
+	i, numBatches := 0, 0
```

Comment:
please use scatter from `mputils` for parallelized requests.
https://github.com/prysmaticlabs/prysm/blob/master/shared/mputil/scatter.go#L21

Reply:
@nisdas do you think that is going to help here? This isn't doing any computation really, just fetching the requests from the server.

Reply:
Mostly because it looks like its doing a lot of repeat work which is done by `Scatter`, its also just splitting up the requests into separate goroutines using a fixed request size.